presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -561,13 +561,13 @@ private PlanNode createRemoteMaterializedExchange(ExchangeNode exchange, Rewrite
         private PartitioningVariableAssignments assignPartitioningVariables(Partitioning partitioning)
         {
             ImmutableList.Builder<VariableReferenceExpression> variables = ImmutableList.builder();
-            ImmutableMap.Builder<VariableReferenceExpression, Expression> constants = ImmutableMap.builder();
+            ImmutableMap.Builder<VariableReferenceExpression, RowExpression> constants = ImmutableMap.builder();
             for (ArgumentBinding argumentBinding : partitioning.getArguments()) {
                 VariableReferenceExpression variable;
                 if (argumentBinding.isConstant()) {
                     ConstantExpression constant = argumentBinding.getConstant();
-                    Expression expression = literalEncoder.toExpression(constant.getValue(), constant.getType());
-                    variable = symbolAllocator.newVariable(expression, constant.getType());
+                    RowExpression expression = constant(constant.getValue(), constant.getType());
```

Comment:
`constant` is already a `RowExpression`; no need to translate.

Reply:
? there isn't translation?

Reply:
I mean `RowExpression expression = constant(constant.getValue(), constant.getType());` is redundant. `constant` is an row expression already. Why break it apart and reassemble?